tests/src/ut_test_control_plane.c

```diff
@@ -31,12 +31,18 @@
 
 #include <ut_control_plane.h>
 
+#include "ut_control_common.h"
```

Comment:
did you reproduce the issue that was raised in the testing sutie? and show that its' now fixed by running the same tests? You should be able to call just the encode and decode functions in that test?

send a single message (message1)
receive that single message (message1) - hold it in a local data
send another message (message2)

Test that your local data does not contain (message2)

Reply:
Yeah, so I have added few changes in the code, so that user can manually see the input and output messages. Test now look like below:

2024-07-25-13:00:48, LOG   , ut_console.c,   639 :      Running Test : 'ut-cp run client'
2024-07-25-13:00:48, DEBUG , ut_control_plane.c,   205 : pthread id 2 = 129370228254272

2024-07-25-13:00:48, LOG   , ut_test_control_plane.c,   272 : Please Run the command `./curl-client-json.sh or/& ./curl-client-yaml.sh or/& ./curl-client-binary.sh` from another terminal and press return;'

2024-07-25-13:00:48, LOG   , ut_test_control_plane.c,   273 : In order to pass the test you need to run each of the curl scripts'

2024-07-25-13:00:51, DEBUG , ut_control_plane.c,   300 : LWS_CALLBACK_HTTP

2024-07-25-13:00:51, DEBUG , ut_control_plane.c,   313 : LWS_CALLBACK_HTTP

2024-07-25-13:00:51, DEBUG , ut_control_plane.c,   316 : LWS_CALLBACK_HTTP, perSessionData not NULL

2024-07-25-13:00:51, DEBUG , ut_control_plane.c,   333 : LWS_CALLBACK_HTTP_BODY_COMPLETION

2024-07-25-13:00:51, DEBUG , ut_control_plane.c,   336 : LWS_CALLBACK_HTTP_BODY_COMPLETION, perSessionData not NULL

2024-07-25-13:00:51, DEBUG , ut_control_plane.c,   222 : DATA RECEIVED

2024-07-25-13:00:51, ERROR , ut_kvp.c,   533 : node not found: UT_KVP_STATUS_KEY_NOT_FOUND
2024-07-25-13:00:51, LOG   , ut_test_control_plane.c,   191 : **************testJSONCallback is called****************
{
  "test": {
    "jsonData": "somevalue",
    "x": 1,
    "on": true
  },
  "test2": {
    "jsonData1": "somevalue1",
    "y": 2,
    "off": false
  }
}

Original Json file
{
  "test": {
    "jsonData": "somevalue",
    "x": 1,
    "on": true
  },
  "test2": {
    "jsonData1": "somevalue1",
    "y": 2,
    "off": false
  }
}

2024-07-25-13:00:52, DEBUG , ut_control_plane.c,   300 : LWS_CALLBACK_HTTP

2024-07-25-13:00:52, DEBUG , ut_control_plane.c,   313 : LWS_CALLBACK_HTTP

2024-07-25-13:00:52, DEBUG , ut_control_plane.c,   316 : LWS_CALLBACK_HTTP, perSessionData not NULL

2024-07-25-13:00:52, DEBUG , ut_control_plane.c,   333 : LWS_CALLBACK_HTTP_BODY_COMPLETION

2024-07-25-13:00:52, DEBUG , ut_control_plane.c,   336 : LWS_CALLBACK_HTTP_BODY_COMPLETION, perSessionData not NULL

2024-07-25-13:00:52, DEBUG , ut_control_plane.c,   222 : DATA RECEIVED
*******************************Inside testYAMLCallback************************
---
test:
  yamlData: somevalue
  x: 1
  on: true
test2:
  yamlData1: somevalue1
  y: 2
  off: false

Original Yaml file
---
test:
  yamlData: somevalue
  x: 1
  on: true
test2:
  yamlData1: somevalue1
  y: 2
  off: false
2024-07-25-13:00:52, ERROR , ut_kvp.c,   533 : node not found: UT_KVP_STATUS_KEY_NOT_FOUND
2024-07-25-13:01:03, LOG   , ut_console.c,   661 :      Test Complete : 'ut-cp run client'

Reply:
[enq-deq-1.c.txt](https://github.com/user-attachments/files/16401648/enq-deq-1.c.txt)
[enq-deq-malloc.c.txt](https://github.com/user-attachments/files/16401649/enq-deq-malloc.c.txt)
As described above wrote a main function to cover the above use case with the changes in PR and without the changes.
In both cases, the tests passed.Please find the tests attached and results below:

> jpn323@janus ~/workspace$ ./with-malloc 
Received message: Hello
Local data does not contain the second message.
jpn323@janus ~/workspace$ 
jpn323@janus ~/workspace$ ./without-malloc
Received message: Hello
Local data does not contain the second message.
jpn323@janus ~/workspace$